jetty-ee11/jetty-ee11-servlet/src/main/java/org/eclipse/jetty/ee11/servlet/ServletChannel.java

```diff
@@ -608,6 +634,268 @@ public void handle()
+                        super.write(b);
+                    }
+                    catch (Throwable t)
+                    {
+                        outputStreamComplete.completeExceptionally(t);
+                        throw t;
+                    }
+                }
+
+                @Override
+                public void write(byte[] b) throws IOException
+                {
+                    try
+                    {
+                        super.write(b);
+                    }
+                    catch (Throwable t)
+                    {
+                        outputStreamComplete.completeExceptionally(t);
+                        throw t;
+                    }
+                }
+
+                @Override
+                public void write(byte[] b, int off, int len) throws IOException
+                {
+                    try
+                    {
+                        super.write(b, off, len);
+                    }
+                    catch (Throwable t)
+                    {
+                        outputStreamComplete.completeExceptionally(t);
+                        throw t;
+                    }
+                }
+
+                @Override
+                public void flush() throws IOException
+                {
+                    try
+                    {
+                        super.flush();
+                    }
+                    catch (Throwable t)
+                    {
+                        outputStreamComplete.completeExceptionally(t);
+                        throw t;
+                    }
+                }
+
+                @Override
+                public void close() throws IOException
+                {
+                    try
+                    {
+                        super.close();
+                        outputStreamComplete.complete(null);
+                    }
+                    catch (Throwable t)
+                    {
+                        outputStreamComplete.completeExceptionally(t);
+                        throw t;
+                    }
+                }
+
+                @Override
+                public void setWriteListener(WriteListener writeListener)
+                {
+                    super.setWriteListener(new WriteListener()
+                    {
+                        @Override
+                        public void onWritePossible() throws IOException
+                        {
+                            writeListener.onWritePossible();
+                        }
+
+                        @Override
+                        public void onError(Throwable t)
+                        {
+                            writeListener.onError(t);
+                            outputStreamComplete.completeExceptionally(t);
+                        }
+                    });
+                }
+            };
+            ServletInputStream inputStream = new ServletInputStreamWrapper(_servletContextRequest.getHttpInput())
```

Comment:
Ditto.
Maybe move both input and output to be inner classes of a ServletUpgrade class that cannot be constructed?